lib/resources/shadow.rb

```diff
@@ -88,38 +82,6 @@ def filter(query = {})
       Shadow.new(@path, content: content, filters: @filters + filters)
     end
 
-    # Next 4 are deprecated methods.  We define them here so we can emit a deprecation message.
-    # They are also defined on the Table, above.
-    def user(query = nil)
-      warn '[DEPRECATION] The shadow `user` property is deprecated and will be removed' \
```

Comment:
So, this was written prior to the `Inspec.deprecate` facility, so this is *currently* acting like a deprecation groups et to `warn`.

If we simply remove these in v4, the user experience will be `no method 'user' for class CLASSx018374 etc...` which is pretty ugly.  Each control will fail, with a source code error.

Alternatively, we could add a deprecation group right now, set it action to warn (for v3) and then set to `exit` in v4, and physically remove only in v5.

That gives a better user experience - the user gets a message that clearly tells them what to do (because we write a nice message) and we also ensure that they actually make the change because the inspec run exits abnormally.

Reply:
... and in fact that is done on https://github.com/inspec/inspec/pull/3829/files#diff-dea4fdbc73cceaa1ff8488720ce40d14 .

I think we should just close this PR.